ext/nnstreamer/tensor_decoder/tensordec-boundingbox.cc

```diff
@@ -0,0 +1,1817 @@
+        bdata = new YoloV5 ();
+        break;
+      case YOLOV8_BOUNDING_BOX:
+        bdata = new YoloV8 ();
+        break;
+      case MP_PALM_DETECTION_BOUNDING_BOX:
+        bdata = new MpPalmDetection ();
+        break;
+      default:
+        return FALSE;
+    }
+  }
+
+  return TRUE;
+}
+
+int
+BoundingBox::setLabelPath (const char *param)
+{
+  if (mode == MP_PALM_DETECTION_BOUNDING_BOX) {
+    /* palm detection does not need label information */
+    return TRUE;
+  }
+
+  if (NULL != label_path)
+    g_free (label_path);
+  label_path = g_strdup (param);
+
+  if (NULL != label_path)
+    loadImageLabels (label_path, &labeldata);
+
+  if (labeldata.total_labels > 0) {
+    bdata->setTotalLabels (labeldata.total_labels);
+    return TRUE;
+  } else
+    return FALSE;
+  /** @todo Do not die for this */
+}
+
+int
+BoundingBox::setVideoSize (const char *param)
+{
+  tensor_dim dim;
+  int rank;
+
+  if (param == NULL || *param == '\0')
+    return TRUE;
+  rank = gst_tensor_parse_dimension (param, dim);
+  width = 0;
+  height = 0;
+
+  if (rank < 2) {
+    GST_ERROR ("mode-option-2 of boundingbox is video output dimension (WIDTH:HEIGHT). The given parameter, \"%s\", is not acceptable.",
+        param);
+    return TRUE; /* Ignore this param */
+  }
+  if (rank > 2) {
+    GST_WARNING ("mode-option-2 of boundingbox is video output dimension (WIDTH:HEIGHT). The third and later elements of the given parameter, \"%s\", are ignored.",
+        param);
+  }
+  width = dim[0];
+  height = dim[1];
+  return TRUE;
+}
+
+int
+BoundingBox::setInputModelSize (const char *param)
+{
+  tensor_dim dim;
+  int rank;
+  if (param == NULL || *param == '\0')
+    return TRUE;
+
+  rank = gst_tensor_parse_dimension (param, dim);
+  bdata->setInputWidth (0);
+  bdata->setInputHeight (0);
+
+  if (rank < 2) {
+    GST_ERROR ("mode-option-3 of boundingbox is input video dimension (WIDTH:HEIGHT). The given parameter, \"%s\", is not acceptable.",
+        param);
+    return TRUE; /* Ignore this param */
+  }
+  if (rank > 2) {
+    GST_WARNING ("mode-option-3 of boundingbox is input video dimension (WIDTH:HEIGHT). The third and later elements of the given parameter, \"%s\", are ignored.",
+        param);
+  }
+  bdata->setInputWidth (dim[0]);
+  bdata->setInputHeight (dim[1]);
+  return TRUE;
+}
+
+int
+BoundingBox::setOption (int opNum, const char *param)
+{
+  if (opNum == 0) {
+    return setBoxDecodingMode (param);
+  } else if (opNum == 1) {
+    return setLabelPath (param);
+  } else if (opNum == 2) {
+    /* option3 = per-decoding-mode option */
+    return bdata->setOptionInternal (param);
+  } else if (opNum == 3) {
+    return setVideoSize (param);
+  } else if (opNum == 4) {
+    return setInputModelSize (param);
+  } else if (opNum == 5) {
+    is_track = (int) g_ascii_strtoll (param, NULL, 10);
+    return TRUE;
+  } else if (opNum == 6) {
+    do_log = (int) g_ascii_strtoll (param, NULL, 10);
+    return TRUE;
+  }
```

Comment:
If they have some meaning, how about making those ```opNum```value constants in the enum class (C++) or enum (C)?

Reply:
Create `BoundingBoxOption` enum class!
Thank you :)